### DISCUSSION TITLE
Is it possible to provide nitro websocket handlers?

### DISCUSSION THREAD

**Discussion by johanalkstal** [solidjs/solid-start#1504] [Q&A] [answered]
Nitro supports acting like a websocket server. The feature is experimental but it can be turned on in solid-start via
```
export default defineConfig({
  server: {
    experimental: {
      websocket: true,
    },
  },
});
```

But now you also need to define a handler to handle those connections in your routes.

https://nitro.unjs.io/guide/websocket

I can't see a way to do this with solid-start api routes. I suspect it's not possible at this time but maybe I'm missing something?

**Comment by hardywu** [comment, marked as answer]
It's doable. Check the doc PR: https://github.com/solidjs/solid-docs-next/pull/760

**Reply by johanalkstal** [reply]
Thanks for the example, much appreciated!
